dotnet/src/SemanticKernel.Abstractions/Search/TextSearchResult.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Represents a search result retrieved from a <see cref="KernelSearchResults{T}" /> instance.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="TextSearchResult"/> class.
+/// </remarks>
+[Experimental("SKEXP0001")]
+public class TextSearchResult(string? name, string? value, string? link, object? innerContent)
```

Comment:
Taking into account, that each search service has its own result model (e.g. `BingWebPage`, Azure AI Search `SearchDocument`, Google `Result`, do we need `TextSearchResult` type?

Reply:
The purpose of this abstraction is to allow for search results to be normalised e.g. if I want to include search results in a prompt I would like to get a normalised result irrespective of the search service I use.